libsql-server/src/rpc/mod.rs

```diff
@@ -122,3 +151,20 @@ fn extract_namespace<T>(
         Err(Status::invalid_argument("Missing x-namespace-bin metadata"))
     }
 }
+
+fn trace_request<B>(req: &hyper::Request<B>, span: &Span) {
+    let _s = span.enter();
+
+    tracing::debug!(
+        "rpc request: {} {} {:?}",
+        req.method(),
+        req.uri(),
+        req.headers()
+    );
+
+    if let Some(v) = req.headers().get("x-libsql-client-version") {
+        if let Ok(s) = v.to_str() {
+            metrics::increment_counter!(CLIENT_VERSION, "version" => s.to_string());
+        }
+    }
+}
```

Comment:
can't we just reuse this function?

Reply:
yeah, we can in a follow up but turns out this was for internal where we don't send any versions so this won't even get called. We prob wanna remove this.